application/pipeline-stacks/common.ts

```diff
@@ -99,11 +102,47 @@ export class PipelineStack extends Stack {
       },
     });
 
-    // Add SSM parameters for Batch task role and profile ARN
+    // Create Lambda function role
+    const lambdaSubmissionRole = new Role(this, `LambdaSubmitRole-${props.workflowName}`, {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMReadOnlyAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+      ]
+    });
+    new Policy(this, `LambdaBatchPolicy-${props.workflowName}`, {
+      roles: [lambdaSubmissionRole],
+      statements: [new PolicyStatement({
+        actions: [
+          'batch:SubmitJob',
+        ],
+        resources: [
+          props.jobQueuePipelineArn,
+          pipelineJobDefinition.jobDefinitionArn,
+        ],
+      })],
+    });
+
+    // Create Lambda function
+    const aws_lambda_function = new LambdaFunction(this, `LambdaSubmissionFunction-${props.workflowName}`, {
+      functionName: `${props.workflowName}-batch-job-submission`,
+      handler: 'lambda_code.main',
+      runtime: Runtime.PYTHON_3_9,
+      code: Code.fromAsset(pathjoin(__dirname, props.workflowName, 'lambda_functions', 'batch_job_submission')),
+      role: lambdaSubmissionRole
+    });
+
+    // Create SSM parameters
```

Comment:
Where are those SSM parameters used?
If the only place is the Lambdas, then perhaps env vars would be better?

Reply:
These SSM parameters are used in the [oncoanalyser](https://github.com/umccr/nextflow-stack/blob/d5b04fb/application/pipeline-stacks/oncoanalyser/assets/run.sh#L227-L257) and [star-align-nf](https://github.com/umccr/nextflow-stack/blob/6b99d9d/application/pipeline-stacks/star-align-nf/assets/run.sh#L107-L125) run scripts.

The one exception is the new `/nextflow_stack/${props.workflowName}/batch_job_definition_arn` parameter, which is currently only accessed by the corresponding job submission Lambdas. I had the same thought that they could be provided as environment variables - seemed to make sense to put as an SSM parameter for consistency since we already have similar data in there (`batch_instance_task_role_arn`, `batch_instance_task_profile_arn`)

Reply:
Hm... not a big deal, but a few points to keep in mind:

- following recommendations we are generally moving away from overly use of SSM parameters in favour of "fixed" configuration. I.e. make the deployment (assets) less dependent on "external configuration options".
- there is a time penalty for using SSM parameters, even more so if used inside the handler code
- there is more "management" overhead as the number of SSM parameters grows

Reply:
Okay, really good to know - I had thought SSM parameters were preferred. I'll avoid using them in the future where there are alternative static configuration options available.